src/compiler/checker.ts

```diff
@@ -22993,6 +23016,10 @@ namespace ts {
                 if (!(links.flags & NodeCheckFlags.ContextChecked)) {
                     links.flags |= NodeCheckFlags.ContextChecked;
                     if (contextualSignature) {
+                        const type = getTypeOfSymbol(getMergedSymbol(node.symbol));
+                        if (isTypeAny(type)) {

```

Comment:
why any? Is that what we get when we call `getTypeOfSymbol` on a symbol that we're already trying to get the type of?

Reply:
Also, this idiom happens three times in this review -- twice more than before I think. Should it be part of some larger chunk of code?

Reply:
`any` because it's good form that when you get an `any` in you make an `any` out. Afaik in normal TS scenarios it's impossible since a new lazy object type is created for each functiony node. The jsdocy codepath there is less well-formed though, and seems perfectly capable of injecting an `any` - so we respect that here and propegate it as expected. I don't think the idiom can be extracted because bit involves control flow, namely a `return`. (And `getTypeOfMergedSymbol()` is almost as long as `getTypeOfSymbol(getMergedSymbol())`, so payoff there is low, imo)

Reply:
:+1: perhaps via `/** @type {any} */` on a function declaration?

It might be true that getTypeOfSymbol should always call getMergedSymbol, but I was referring to `if(any)`.